### PR TITLE
[no ticket][risk=no] Puppeteer notebook test

### PR DESCRIPTION
**Pull request by aweng98**
Notebook basic tests in Python 3 and R language.
- Three new tests: `notebook-r.spec.ts`, `notebook-python3.spec.ts` and `notebook-actions.spec.ts`
- Renamed `analysis-page` to `workspace-analysis-page` file name.
- Unrelated to the new tests, fixes for flaky functions

`tests/cohorts/cohorts-rename.spec.ts` is failing due to a UI change. Test code will be addressed in new PR.

### CODE REVIEW

File: e2e/app/component/new-notebook-modal.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import {waitForText} from 'utils/waits-utils';
+import RadioButton from 'app/element/radiobutton';
+import {Language, LinkText} from 'app/text-labels';
+import Button from 'app/element/button';
+import Textbox from 'app/element/textbox';
+import Modal from './modal';
+
+const modalTitle = 'New Notebook';
+
+export default class NewNotebookModal extends Modal {
```
Comment (NehaBroad) [inline e2e/app/component/new-notebook-modal.ts:12]: Nice! Can we add a test for error like if the notebook name already exist?

File: e2e/app/page/notebook-page.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    }
+    // Press Enter key to activate edit mode
+    return this.page.keyboard.press('Enter');
+  }
+
+  /**
+   * Run (Execute) cells using Shift+Enter keys.
+   * Shortcut explanation: Shift+Enter run cells and select below.
+   */
+  async runCellShiftEnter(): Promise<void> {
```
Comment (NehaBroad) [inline e2e/app/page/notebook-page.ts:150]: Optional: We can combine these three methods by just passing 'Shift'/'Control'. 'Alt' and replacing them in keyboard.down. Or 
Keep these three methods and have each one call a common method that takes the command and do the following:
await this.page.bringToFront();
    await this.page.keyboard.down(command);
    await this.page.keyboard.press('Enter', {delay: 20});
    await this.page.keyboard.up(command);

File: e2e/app/page/notebook-page.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  Command= 'command_mode',
+  Edit = 'command_mode',
+}
+
+export enum KernelStatus {
+  notRunning = 'Kernel is not running',
+  Idle = 'Kernel Idle',
+}
+
+export default class NotebookPage extends AuthenticatedPage {
```
Comment (NehaBroad) [inline e2e/app/page/notebook-page.ts:28]: Can we add a test to make sure that on clicking any Download Option from File menu we see a pop Up with AoU policy

Comment (aweng98) [reply e2e/app/page/notebook-page.ts:30]: I'll add this test in new PR next time.

File: e2e/utils/test-utils.ts [hunk 4]
```diff
@@ -45,36 +45,24 @@ export async function signInAs(userId: string, passwd: string): Promise<Page> {
...
-  let spinner: JSHandle;
-  try {
-    spinner = await page.waitFor((selector) => {
-      return document.querySelectorAll(selector).length > 0
-    }, {timeout: 1000}, spinSelector);
-  } catch (err) {
-    console.info('waitWhileLoading does not find any spin elements.');
-  }
-  const jValue = await spinner.jsonValue();
+export async function waitWhileLoading(page: Page, timeOut?: number): Promise<void> {
```
Comment (aweng98) [inline e2e/utils/test-utils.ts:48]: reworked `waitWhileLoading` function

**Pull request merged**